Add Sphinx documentation and GitHub Pages workflow

1597463007 opened the pull request:
Introduces Sphinx-based documentation for the project, including configuration, requirements, and initial content for adapter and protocol documentation. Adds a GitHub Actions workflow to build and publish the documentation to GitHub Pages on release or manual trigger. Also includes .gitignore updates to exclude build artifacts and IDE files.

1597463007 committed:
Add Sphinx documentation and GitHub Pages workflow

Introduces Sphinx-based documentation for the project, including configuration, requirements, and initial content for adapter and protocol documentation. Adds a GitHub Actions workflow to build and publish the documentation to GitHub Pages on release or manual trigger. Also includes .gitignore updates to exclude build artifacts and IDE files.

1597463007 committed:
Add missing imports to scaling controller interface

rafa-be commented on docs/source/_static/schema.yaml:
```diff
@@ -0,0 +1,86 @@
+        401:
+          description: Unauthorized, e.g., invalid or missing auth_token
+          schema:
+            $ref: '#/definitions/Error'
```
We probably need a 404 error for unknown `worker_id`s.

sharpener6 commented on docs/source/adapter/webhook.rst:
```diff
@@ -0,0 +1,40 @@
+                                            │          └────────┘
+                                            │                    
+    ┌───────────┐ Webhook Requests  ┌───────┴─┐ Start  ┌────────┐
+    │  Scaling  ├───────────────────► Adapter ├────────► Worker │
```
also this chart might be better went to the summary page

1. summary:
    - user need implement allocate policy or using existing ones
    - user need implement worker manager adapters or using existing ones
    - then chart
    - briefly explain the roles of above
Then you have 2 separate pages talks about
1. how allocate policy is implemented in detail
2. how worker manager adapter is implemented, also include detail about json formatting etc.

1597463007 committed:
Expand scaling and webhook adapter documentation

1597463007 committed:
Update scaling docs to use worker groups

Replaces references to individual workers with worker groups in schema.yaml and documentation. Updates webhook request and response examples, handler method names, and scaling controller implementation notes to reflect the new worker group model.
